odoo/addons/test_lint/tests/_odoo_checker_sql_injection.py

```diff
@@ -51,7 +51,7 @@ def _allowable(self, node):
             and isinstance(node.expr, astroid.Name)
             and node.attrname.startswith('_')
             # cr.execute('SELECT * FROM %s' % 'table') is OK since that is a constant
-            or isinstance(node, astroid.Const)
+            or isinstance(infered, astroid.Const)
```

Comment:
Should be moved up to the `infered` guard at line 44 (guard which should probably be split / rewritten to better handle multiple conditions in case the inference succeeded).

Reply:
@xmo-odoo I'm checking this a bit more in-depth because this change allows some BinOp to bypass the end of _check_concatenation() by validating the condition on line 60. But if the result is correct it should be the same